src/app/services/baw-api/base-api.service.spec.ts

```diff
@@ -1,68 +1,855 @@
+      signIn("xxxxxxxxxxxxxxx", "username");
+      signOut();
+      expect(service.getSessionUser()).toBe(null);
+    });
   });
 
-  it("should be logged in after user saved to session storage", () => {
-    const user = new SessionUser({
-      authToken: "aaaaaaaaaaaaaaaaaaaaaa",
-      userName: "username"
+  describe("HTTP Request Methods", () => {
+    const httpMethods = [
+      {
+        functionName: "httpGet",
+        method: "GET",
+        hasBody: false
+      },
+      {
+        functionName: "httpPost",
+        method: "POST",
+        hasBody: true
+      },
+      {
+        functionName: "httpPatch",
+        method: "PATCH",
+        hasBody: true
+      },
+      {
+        functionName: "httpDelete",
+        method: "DELETE",
+        hasBody: false
+      }
+    ];
+
+    httpMethods.forEach(httpMethod => {
+      describe(httpMethod.functionName, () => {
+        it("should create patch request", () => {
+          service[httpMethod.functionName]("/broken_link").subscribe();
+          const req = catchRequest("/broken_link", httpMethod.method);
+          expect(req).toBeTruthy();
+        });
+
+        it("should create request with baw server headers", () => {
+          service[httpMethod.functionName]("/broken_link").subscribe();
+          const req = catchRequest("/broken_link", httpMethod.method);
+          verifyHeaders(req);
+        });
+
+        it("should create request with authenticated baw server headers", () => {
+          signIn("xxxxxxxxxxxxxxx", "username");
+          service[httpMethod.functionName]("/broken_link").subscribe();
+          const req = catchRequest("/broken_link", httpMethod.method);
+          verifyAuthHeaders(req, "xxxxxxxxxxxxxxx");
+        });
+
+        it("should return single response", () => {
+          const response = {
+            meta: singleMeta,
+            data: singleResponse
+          } as ApiResponse<MockModel>;
+
+          service[httpMethod.functionName]("/broken_link", {}).subscribe(
+            (data: ApiResponse<MockModel>) => expect(data).toEqual(response),
+            shouldNotFail
+          );
+
+          const req = catchRequest("/broken_link", httpMethod.method);
+          flushResponse<MockModel>(req, response);
+        });
+
+        it("should return multi response", () => {
+          const response = {
+            meta: singleMeta,
+            data: [singleResponse]
+          } as ApiResponse<MockModel[]>;
+
+          service[httpMethod.functionName]("/broken_link", {}).subscribe(
+            (data: ApiResponse<MockModel[]>) => expect(data).toEqual(response),
+            shouldNotFail
+          );
+
+          const req = catchRequest("/broken_link", httpMethod.method);
+          flushResponse<MockModel[]>(req, response);
+        });
+
+        it("should handle error response", () => {
+          const response = { meta: errorMeta, data: null } as ApiResponse<
+            MockModel
+          >;
```

Comment:
you're line length linting rules seem a bit aggressive

Reply:
Took a crack at fixing it, Prettier (my formatter) has options to expand the line length but its ignoring whatever i set to ill have to fix it another time